src/util/InstallDriver.ts

```diff
@@ -495,10 +498,48 @@ class InstallDriver {
     this.mApi.dismissNotification(getUnfulfilledNotificationId(collection.id));
     this.mApi.store.dispatch(actions.setModEnabled(profile.id, collection.id, true));
 
-    const required = (collection?.rules ?? [])
+    const required = []
       .filter(rule => rule.type === 'requires');
-    this.mRequiredMods = required
-      .filter(rule => util.findModByRef(rule.reference, mods) === undefined);
+    for (const rule of (collection?.rules ?? [])) {
+      if (rule.type !== 'requires') {
+        continue;
+      }
+
+      const modByRef = util.findModByRef(rule.reference, mods);
+      if (modByRef === undefined) {
+        required.push(rule);
+        continue;
+      }
+
+      if (rule.fileList?.length ?? 0 > 0) {
+        const stagingPath = selectors.installPathForGame(state, profile.gameId);
+        const modPath = path.join(stagingPath, modByRef.installationPath);
+        const presentFiles = await new Promise(async (resolve, reject) => {
+          let entries: IEntry[];
+          await turbowalk(modPath, async input => {
+            entries = [].concat(entries, input);
+          }, {});
+
+          const presentFiles = await Promise.all(entries
+            .filter(iter => !!iter && !iter.isDirectory)
+            .map(async iter => ({
+              path: path.relative(modPath, iter.filePath),
+              md5: await fileMD5Async(iter.filePath),
+            })
+          ));
+          return resolve(presentFiles);
+        });
+        if (!_.isEqual(presentFiles, rule.fileList)) {
```

Comment:
For one thing: You can't rely on the sort order of these files to be the same as in the file list, neither of these lists is sorted. It could be the exact same list of files but in a different order or - in case of a case insensitive filesystem, different casing. The fileList in the collection could have come from a completely different file system.

And then this also seems inefficient. You hash the entire file list, then compare it, I think it would be better you just go through the files and break out of the loop the moment you find the first non-matching file.
That way in case the files _are_ different, we don't have to hash them all.